src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -53,3 +54,86 @@
      handle-connection)))
 
 (rf/reg-event-fx :wallet/get-key-pair-export-connection get-key-pair-export-connection)
+
+(defn extract-keypair-name
+  [db key-uids-set]
+  (when (= (count key-uids-set) 1)
+    (let [key-uid  (first key-uids-set)
+          keypairs (get-in db [:wallet :keypairs])]
+      (->> (filter #(= (:key-uid %) key-uid) keypairs)
+           first
+           :name))))
+
+(defn update-accounts
```

Comment:
An idea you may like is to consider putting functions operating on domain data in a `data-store` namespace. Like the namespace `contexts.wallet.data-store`. A few of these data store namespaces are still inside `src/legacy/`, but you could create another file just for the wallet settings domain.

Architecturally, it makes sense to me because we can avoid tying ourselves 100% with re-frame.

Reply:
Just checked it, that's nice and more clear. will do